### ISSUE TITLE
[YCQL] Unexpected results with TTL

### ISSUE THREAD

**Issue by kcajmagic** [yugabyte/yugabyte-db#2091]
# Description
I was playing around with TTL from the [docs](https://docs.yugabyte.com/latest/architecture/docdb/persistence/#update-row-with-ttl)
When you update a row with a greater time to live, the row doesn't have the expected data

## Steps to reproduce the issue:
Using `cqlsh -C`

```cql
CREATE KEYSPACE site;
CREATE TABLE site.page_views (page_id text, views int, category text, PRIMARY KEY ((page_id)));
INSERT INTO site.page_views (page_id, views) VALUES ('abc.com', 10)  USING TTL 5;
SELECT * FROM site.page_views;
UPDATE site.page_views USING TTL 10 SET category = 'news' WHERE page_id = 'abc.com';
SELECT * FROM site.page_views;
```

## Results Received:
- After Insert 
```
 page_id | views | category
---------+-------+----------
 abc.com |    10 |     null
```
- After Insert  +10s
```
 page_id | views | category
---------+-------+----------
 abc.com |    10 |     null
```
- After Update
```
 page_id | views | category
---------+-------+----------
 abc.com |  null |     news
```
- After Update + 10s
```
 page_id | views | category
---------+-------+----------
 abc.com |  null |     news
```

## Results Expected
- After Insert 
```
 page_id | views | category
---------+-------+----------
 abc.com |    10 |     null
```
- After Insert  +10s
```
 page_id | views | category
---------+-------+----------

```
- After Update
depending on the time after the initial command I would expect different results.
  - if update is within 5s I would expect:
```
 page_id | views | category
---------+-------+----------
 abc.com |    10 |     news
```
  - if update was greater than 5s I would expect an error since row does not exist with  `page_id = 'abc.com'`

## Versions used
yugabte: 1.3.1
cqlsh: 5.0.1
os: osx 10.14.6

**Comment by kmuthukk** [comment]
hi @kcajmagic :

Interesting. I am not able to reproduce the INSERT +10s issue. In my 1.3.1 setup the row is disappearing as expected:

```
cqlsh> CREATE KEYSPACE site;
cqlsh> CREATE TABLE site.page_views (page_id text, views int, category text, PRIMARY KEY ((page_id)));
cqlsh> INSERT INTO site.page_views (page_id, views) VALUES ('abc.com', 10)  USING TTL 5;
cqlsh> SELECT * FROM site.page_views;

 page_id | views | category
---------+-------+----------
 abc.com |    10 |     null

(1 rows)

... after about 10 seconds ...

cqlsh> SELECT * FROM site.page_views;

 page_id | views | category
---------+-------+----------

(0 rows)
```

**Comment by kmuthukk** [comment]
Also, see this. This is also seems as expected.

```
cqlsh> INSERT INTO site.page_views (page_id, views) VALUES ('abc.com', 10)  USING TTL 10;
cqlsh> UPDATE site.page_views USING TTL 25 SET category = 'news' WHERE page_id = 'abc.com';

cqlsh> SELECT ttl(category) FROM site.page_views;

 ttl(category)
---------------
            21

(1 rows)
cqlsh> SELECT ttl(category) FROM site.page_views;

 ttl(category)
---------------
            15

(1 rows)
cqlsh> SELECT ttl(category) FROM site.page_views;

 ttl(category)
---------------
             8

(1 rows)
cqlsh> SELECT ttl(category) FROM site.page_views;

 ttl(category)
---------------

(0 rows)
```

**Comment by kcajmagic** [comment]
Thank you that `ttl` query is awesome.

Unfortunately locally I am seeing:
```
cqlsh> CREATE TABLE site.page_views (page_id text, views int, category text, PRIMARY KEY ((page_id)));
cqlsh> INSERT INTO site.page_views (page_id, views) VALUES ('abc.com', 10)  USING TTL 5;
cqlsh> SELECT ttl(views) FROM site.page_views where page_id='abc.com';

 ttl(views)
------------
          5

(1 rows)
cqlsh> SELECT ttl(views) FROM site.page_views where page_id='abc.com';

 ttl(views)
------------
          5

(1 rows)
cqlsh> SELECT ttl(views) FROM site.page_views where page_id='abc.com';

 ttl(views)
------------
          5

(1 rows)
cqlsh> SELECT ttl(views) FROM site.page_views where page_id='abc.com';

 ttl(views)
------------
          5

(1 rows)
cqlsh> SELECT ttl(views) FROM site.page_views where page_id='abc.com';

 ttl(views)
------------
          5

(1 rows)
```

**Comment by kmuthukk** [comment]
wow... that's definitely odd. 

I was trying on CentOS. Will try on Mac OSX when I get a chance. Otherwise, we might need your help to try and look at your environment over a conf call.

@m-iancu - any other ideas?

**Comment by m-iancu** [comment]
@kcajmagic Can you retry the statements above but update the `SELECT` query as follows:
```
 SELECT ttl(views), writetime(views), toUnixTimestamp(now()) FROM site.page_views where page_id='abc.com';
```
And share the output.

**Comment by kcajmagic** [comment]
```
cqlsh>  SELECT ttl(views), writetime(views), toUnixTimestamp(now()) FROM site.page_views where page_id='abc.com';

 ttl(views) | writetime(views) | tounixtimestamp(now())
------------+------------------+------------------------
          5 | 1565993648108999 |          1565993656425

(1 rows)
```

**Comment by kmuthukk** [comment]
I was not able to reproduce the reported issue on Mac OSX either with v1.3.1. 

@kcajmagic -- can you make sure to try after a clean drop table? If problem is still happening, let us know how we can look further at your system to debug the issue.

**Comment by m-iancu** [comment]
@kcajmagic Could you also share more information about the deployment scenario?
Assuming this is a local cluster did you use any specific arguments to `./bin/yb-ctl create`?
You can also check the status of the node(s) with `./bin/yb-ctl status` or by going to the master UI (e.g. `http://127.0.0.1:7000/tablet-servers` if `127.0.0.1` is the master leader) to make sure the cluster is in a healthy state.

**Comment by kmuthukk** [comment]
This is quite likely a duplicate of #2394. 

@kcajmagic -- Is it possible you are using a version of Apache Cassandra version of cqlsh where the default consistency is weaker. Can you test with https://github.com/yugabyte/cqlsh?

In any case, this is a bug on our side (when weaker read consistency is use). But as noted in #2394 impact should be minimal. The enforcement appears to be a bit delayed until some new writes have happened.

**Comment by kmuthukk** [comment]
@kcajmagic -- this has been fixed in #2394.

This turned out to be specific to Replication Factor = 1 AND weaker consistency level read being used.
